app/Models/Asset.php

```diff
@@ -177,6 +175,51 @@ public function checkOut($target, $admin = null, $checkout_at = null, $expected_
         return false;
     }
 
+    public function checkIn($target, $item_name = null, $status_id = null)
+    {
+        $admin = Auth::user();
+        if ($this->assignedType() == Asset::USER) {
+            $user = $this->assignedTo;
+        }
+
+        $this->expected_checkin = null;
+        $this->last_checkout = null;
+        $this->assigned_to = null;
+        $this->assignedTo()->disassociate($this);
+        $this->assigned_type = null;
+        $this->accepted = null;
+        if (isset($item_name)) {
+            $this->name = $item_name;
+        }
+
+        // Only update this if one is passed, otherwise we'll null it out.
+        if ($status_id) {
+            $this->status_id =  $status_id;
+        }
+        // Was the asset updated?
+        if ($this->save()) {
+            $logaction = $this->logCheckin($target, e(request('note')));
+
+            $data['log_id'] = $logaction->id;
+            $data['first_name'] = get_class($target) == User::class ? $target->first_name : '';
+            $data['item_name'] = $this->present()->name();
+            $data['checkin_date'] = $logaction->created_at;
+            $data['item_tag'] = $this->asset_tag;
+            $data['item_serial'] = $this->serial;
+            $data['note'] = $logaction->note;
+
+            if ((($this->checkin_email()=='1')) && (isset($user)) && (!empty($user->email)) && (!config('app.lock_passwords'))) {
+                Mail::send('emails.checkin-asset', $data, function ($m) use ($user) {
+                    $m->to($user->email, $user->first_name . ' ' . $user->last_name);
+                    $m->replyTo(config('mail.reply_to.address'), config('mail.reply_to.name'));
+                    $m->subject(trans('mail.Confirm_Asset_Checkin'));
+                });
+            }
+            return true;
```

Comment:
Should we try to use the notifiables stuff here instead?

Reply:
In a perfect world, yes.  Given how messy the transition was for checkout, I wanted to do it as a separate change.